### PR TITLE
feat(dashboard): make nono sandbox synthesis work for any agent via config

### LINKED ISSUE #146: feat(dashboard): add nono/macOS sandbox support for zai agent
## Problem

On macOS, `zai` cannot run as a sandboxed agent in lince-dashboard. The plugin's `synthesize_sandboxed_command` function has no match arm for `"zai"`, so it falls through to `_ => None` and falls back to the static `command` field — which references `agent-sandbox`, a Linux-only binary that doesn't exist on macOS.

## Why it works on Linux but not macOS

- **Linux**: Uses `agent-sandbox` backend. The static `command` field in config.toml already contains the full wrapper: `["agent-sandbox", "run", "-p", "{project_dir}", "--id", "{agent_id}", "--agent", "zai"]`. The plugin runs it as-is — no synthesis needed.
- **macOS**: Uses `nono` backend (agent-sandbox/bwrap not available). The plugin must synthesize a `nono run --profile lince-zai-permissive -- zai` command. This requires knowing the inner binary name and home directory, which comes from the match arm lookup.

## Fix

1. Add `"zai" => (vec!["zai".to_string()], ".cc-mirror/zai")` match arm in `agent.rs`
2. Add `paolo_config.sh` — setup script that installs nono profiles and writes config.toml for zai as default sandboxed agent

## Testing

- [ ] Pull `feature/zai-nono-sandbox` branch on Mac
- [ ] Rebuild WASM plugin: `cd lince-dashboard/plugin && ./build.sh`
- [ ] Reinstall: `cd lince-dashboard && ./install.sh`
- [ ] Run `bash paolo_config.sh` to install nono profiles
- [ ] Restart zellij and launch `lince-dashboard`
- [ ] Press `[n]` — zai should appear as default, sandboxed
- [ ] Press `[N]` — wizard should show zai with sandbox level options

## Architectural Note

Every new agent type currently requires a Rust code change + WASM rebuild because the match arm is hardcoded. A config-driven approach (deriving inner binary from agent type name, home dir from `home_rw_dirs`) would eliminate this friction.

### PR DESCRIPTION
**Pull request by paoloantinori**
## Summary
- **Config-driven nono synthesis**: unknown agent types can now use nono sandboxing via configuration alone — no code changes needed
- **New `sandbox_home_subdir` config field**: optional field for paranoid nono mode's rsync scratch directory
- **Config parsing robustness**: `unwrap_or_default()` → explicit `match` for TOML parse errors

## The Problem

`synthesize_sandboxed_command()` had a hardcoded match arm per agent type. When an unknown agent hit `_ => return None`, the plugin silently fell back to the static `command` template — **ignoring `sandbox_backend = "nono"` entirely**. Any agent not in the hardcoded list could not use nono sandboxing, even though its config said it should.

## The Fix

The `_` fallback now uses the agent's `[agents.<name>].command` from config as the inner command for nono/none synthesis. This means:

1. Add a `[agents.my-agent]` section in config with `sandbox_level` and `sandbox_backend = "nono"`
2. Provide a matching nono profile at `~/.config/nono/profiles/lince-my-agent.json`
3. Done — no plugin code changes needed

Existing hardcoded agents (claude, codex, gemini, opencode, pi) are **completely unchanged** — zero regression risk.

## Files Changed

| File | Change |
|---|---|
| `agent.rs` | Add `type_config` param to `synthesize_sandboxed_command`, config-driven `_` fallback |
| `config.rs` | Add `sandbox_home_subdir` field, `unwrap_or_default` → `match` fix, 3 new tests |

## How it works for a custom agent (e.g. zai)

```toml
# config.toml
[agents.zai]
command = ["zai"]
pane_title_pattern = "agent-sandbox"
status_pipe_name = "claude-status"
display_name = "ZAI"
short_label = "ZAI"
color = "blue"
sandboxed = true
has_native_hooks = true
providers = ["__discover__"]
sandbox_backend = "nono"
sandbox_level = "permissive"
sandbox_home_subdir = ".cc-mirror/zai"
```

With a nono profile at `~/.config/nono/profiles/lince-zai-permissive.json`, the plugin synthesizes:
```
nono run --profile lince-zai-permissive --workdir /my/project -- zai
```

## Test plan
- [x] WASM build (`wasm32-wasip1`) passes
- [x] Config parsing tests for new `sandbox_home_subdir` field
- [x] Malformed TOML handling test (`unwrap→match` fix)
- [ ] Manual verification with a custom agent + nono profile
- [ ] Verify existing agents (claude, codex, gemini) still work unchanged

Closes #146

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lince-dashboard/plugin/src/agent.rs [hunk 1]
```diff
@@ -240,7 +241,21 @@ fn synthesize_sandboxed_command(
         "bash" => (vec!["bash".to_string()], ""),
         "zsh" => (vec!["zsh".to_string()], ""),
         "fish" => (vec!["fish".to_string()], ""),
-        _ => return None,
+        // Config-driven fallback: unknown agents use their [agents.<name>].command
+        // as the inner command for nono/none synthesis. Requires a matching
+        // nono profile at ~/.config/nono/profiles/lince-<agent>[-<level>].json.
+        _ => match type_config {
+            Some(cfg) => {
```
Comment (maeste) [inline lince-dashboard/plugin/src/agent.rs:248]: ✅ Clean use of expand_command_template for the config-driven inner command. The template expansion handles {agent_id}, {project_dir} etc. consistently with known agents.

File: lince-dashboard/plugin/src/agent.rs [hunk 2]
```diff
@@ -240,7 +241,21 @@ fn synthesize_sandboxed_command(
...
+                let inner = expand_command_template(
+                    &cfg.command, agent_id, project_dir, None,
+                );
+                let home_sub = cfg.sandbox_home_subdir
+                    .as_deref()
```
Comment (maeste) [inline lince-dashboard/plugin/src/agent.rs:253]: 💡 **Minor:** sandbox_home_subdir defaults to empty string for unknown agents. This means paranoid mode will skip the rsync step (clean HOME), which is appropriate for stateless agents but may surprise someone expecting automatic home-dir detection. The doc comment clarifies this, so acceptable.

File: lince-dashboard/plugin/src/config.rs [hunk 3]
```diff
@@ -705,9 +712,10 @@ pub fn parse_agent_defaults(stdout: &[u8]) -> HashMap<String, AgentTypeConfig> {
     // Parse defaults file (chunk 0) — uses [agents.<name>] structure.
     let mut agent_types: HashMap<String, AgentTypeConfig> = if let Some(chunk) = chunks.first() {
         let content = String::from_utf8_lossy(chunk);
-        toml::from_str::<UserAgentsSection>(content.trim())
-            .map(|s| s.agents)
-            .unwrap_or_default()
+        match toml::from_str::<UserAgentsSection>(content.trim()) {
```
Comment (maeste) [inline lince-dashboard/plugin/src/config.rs:715]: ✅ Good: replacing unwrap_or_default() with explicit match arms. Malformed TOML now returns HashMap::new() explicitly rather than relying on serde's default, which could mask parse errors.

File: lince-dashboard/plugin/src/config.rs [hunk 4]
```diff
@@ -1097,4 +1105,58 @@ mod tests {
...
+    #[test]
+    fn unknown_agent_type_with_sandbox_home_subdir_parses() {
+        let toml_text = r#"
+[agents.custom-agent]
+command = ["custom-agent", "--flag"]
+pane_title_pattern = "custom-agent"
+status_pipe_name = "custom-status"
+display_name = "Custom Agent"
+short_label = "CST"
+color = "green"
```
Comment (maeste) [inline lince-dashboard/plugin/src/config.rs:1120]: ✅ Test coverage for the new field is complete: presence, absence, and malformed TOML edge case all covered.

**Review by maeste — APPROVED** [review]
## Review: PR #148 — feat(dashboard): make nono sandbox synthesis work for any agent via config

### Verdict: **Approve** — well-designed feature, good test coverage, zero regression risk for existing agents.

---

### ✅ Looks Good

- The config-driven fallback for unknown agent types is the right architectural choice. Instead of adding hardcoded match arms per new agent, users can now define custom agents entirely via TOML config + nono profile.
- The `_ => match type_config { Some(cfg) ... None => return None }` pattern preserves backward compatibility: existing agents are untouched, unknown agents without config still return None.
- New `sandbox_home_subdir` field is optional with `#[serde(default)]`, fully backward-compatible.
- `unwrap_or_default()` → explicit `match` for TOML parse errors is a correctness improvement (malformed TOML no longer silently produces an empty map that looks like valid data).
- Test coverage is thorough: 3 new tests covering happy path, default value, and malformed input edge case.
- Doc comments on `sandbox_home_subdir` clearly explain when it is relevant (nono + paranoid only).

### 💡 Suggestions

- **`expand_command_template` call in the fallback**: The fallback uses `expand_command_template(&cfg.command, agent_id, project_dir, None)`. The `None` for what appears to be an optional type_config-like param is worth confirming intentional — it means recursive template expansion won't have access to agent-specific config context. Likely fine since this is the unknown-agent path.
- **Consider documenting the required nono profile convention** somewhere visible (README or docs): users need to know profiles must follow the `lince-<agent>[-<level>].json` naming pattern. The PR body mentions it, but a persistent doc reference would help discoverability.

**Comment by maeste** [comment]
## Code Review Summary

**Verdict: ✅ Approved** (0 blocking, 0 warnings, 2 suggestions)

### ✅ Looks Good
- Config-driven fallback for unknown agents is the right architectural choice
- `_ => match type_config` preserves full backward compatibility (existing agents untouched)
- `sandbox_home_subdir` is optional with serde default, fully backward-compatible
- `unwrap_or_default()` → explicit `match` for TOML parse correctness
- 3 new tests: happy path, default value, malformed edge case
- Doc comments are clear about when the field is relevant (nono + paranoid only)

### 💡 Suggestions (non-blocking)
- Consider documenting the nono profile naming convention (`lince-<agent>[-<level>].json`) in persistent docs for discoverability
- The `None` param in `expand_command_template` call is worth confirming intentional (no recursive agent config context)

---
*Reviewed by Hermes Agent*
